sdk/extensions/Microsoft.Extensions.Azure/tests/AzureClientFactoryTests.cs

```diff
@@ -417,9 +450,302 @@ public void CanRegisterCustomClientWithOptionsAndCredentialProvider()
+            var client = factory.CreateClient(nameof(AsyncDisposableClient));
+
+            provider.Dispose();
+
+            using var cancellationSource = new CancellationTokenSource(TimeSpan.FromSeconds(5));
+
+            Assert.DoesNotThrowAsync(async () => await tcs.Task.AwaitWithCancellation(cancellationSource.Token));
+            Assert.IsTrue(disposed);
+        }
+
+        [Test]
+        public void DisposeHandlesNonDisposableClients()
+        {
+            var services = new ServiceCollection();
+
+            services.AddAzureClients(builder =>
+                builder.AddClient<NonDisposableClient, object>(_ => new NonDisposableClient()).WithName(nameof(NonDisposableClient)));
+
+            var provider = services.BuildServiceProvider();
+            var factory = provider.GetRequiredService<IAzureClientFactory<NonDisposableClient>>();
+            var client = factory.CreateClient(nameof(NonDisposableClient));
+
+            Assert.DoesNotThrow(provider.Dispose);
+        }
+
+        [Test]
+        public void DisposableHandlesMixedClients()
+        {
+            var tcs = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
+            var services = new ServiceCollection();
+            var disposeCount = 0;
+
+            Action disposeCallback= () =>
+            {
+                if (++disposeCount >= 3)
+                {
+                    tcs.TrySetResult(true);
+                }
+            };
+
+            services.AddAzureClients(builder =>
+                builder.AddClient<NonDisposableClient, object>(_ => new NonDisposableClient()).WithName(nameof(NonDisposableClient)));
+
+            services.AddAzureClients(builder =>
+                builder.AddClient<BothDisposableClient, object>(_ => new BothDisposableClient(disposeCallback)).WithName(nameof(BothDisposableClient)));
+
+            services.AddAzureClients(builder =>
+                builder.AddClient<AsyncDisposableClient, object>(_ => new AsyncDisposableClient(disposeCallback)).WithName(nameof(AsyncDisposableClient)));
+
+            services.AddAzureClients(builder =>
+                builder.AddClient<DisposableClient, object>(_ => new DisposableClient(disposeCallback)).WithName(nameof(DisposableClient)));
+
+            var provider = services.BuildServiceProvider();
+
+            var factory = provider.GetRequiredService<IAzureClientFactory<NonDisposableClient>>();
+            var client = factory.CreateClient(nameof(NonDisposableClient));
+
+            var disposableFactory = provider.GetRequiredService<IAzureClientFactory<DisposableClient>>();
+            var disposableClient = disposableFactory.CreateClient(nameof(DisposableClient));
+
+            var asyncDisposableFactory = provider.GetRequiredService<IAzureClientFactory<AsyncDisposableClient>>();
+            var asyncDispsableClient = asyncDisposableFactory.CreateClient(nameof(AsyncDisposableClient));
```

Comment:
typo 'asyncDispsableClient'  should be 'asyncDisposableClient' instead.

Reply:
Curious why cspell did not check it.

Reply:
I don't think cspell checks variable names, as even with the correct spelling it's still not an actual word.   

<img src="https://media2.giphy.com/media/ChmEWOL7Vaz5u/giphy.gif"/>